### ISSUE TITLE
feature: "search" functionality over files  

### ISSUE THREAD

**Issue by adamdecaf** [moov-io/ach-test-harness#37]
Often during testing a developer wants to retrieve the file / batches / entries that are uploaded. This is done to verify accurate generation of the files. With the nature of ACH and filesystems this can be a bit complicated because tests end up needing to read a directory of files (e.g. `outbound`), parse and filter them in order to find what a specific test is looking for.

Perhaps we could offer a search endpoint to streamline this? ach-test-harness can inspect outbound/returned/etc directories and return the files in search. This also allows for the underlying filesystem to be hidden (e.g. if ach-test-harness is running on a different machine).

**Comment by adamdecaf** [comment]
An example I'm thinking about:

```
GET /search?dirs=outbound,returned&traceNumber=XXXX

GET /search?dirs=outbound,returned&RDFIIdentification=YYYY
```

```
{
  "files": [
    {
      "filepath": "outbound/foo.ach",
      "file": { ... }
    }  
  ]
}
```

**Comment by adamdecaf** [comment]
Initially I see the follow query params to search on:
- RDFIIdentification / CheckDigit (ABA 8 vs ABA 9)
- DFIAccountNumber
- Amount

FYI this matching can hopefully use the existing matching logic. It seems very similar (and was hopefully designed to support both types of matching logic). 

https://github.com/moov-io/ach-test-harness/blob/master/pkg/response/matcher.go

**Comment by alovak** [comment]
I was thinking more about ... RESTy endpoints like:
```
GET /transactions?routingNumber=xxx&accountNumber:xxx

[
  {
    "id": "transactionID",
    "description": "test transaction",
    "companyIdentification": "moov",
    "amount": "xxx",
    "type": "credit",
    ...
    }
 ]
```

**Comment by alovak** [comment]
but maybe mine is too specific, while we should support queries for all/different ACH fields...

**Comment by adamdecaf** [comment]
I like `/transactions`. It can return just the `*ach.EntryDetail` record. I was thinking returning the entire `*ach.File` would be annoying to filter through.

**Comment by alovak** [comment]
I imagine this use case:
```go

// in test
transactions, err := achHarness.ListTransactions(achHarness.ListOptions{...})
```

**Comment by adamdecaf** [comment]
Yep. That looks great!

**Comment by adamdecaf** [comment]
I realized that ach-test-harness's main "server" is FTP. It only listens on HTTP for the admin server. I'm okay putting this on the admin server so we don't have to initialize a third port / server.

**Comment by alovak** [comment]
Cool! I'll work on it.

**Comment by adamdecaf** [comment]
I've released v0.5.0 with this included, thanks! 

 https://github.com/moov-io/ach-test-harness/releases/tag/v0.5.0
